api/src/controllers/publications.js

```diff
@@ -150,4 +154,89 @@ async function readAllPublicationsByTeam(req, res) {
+            },
+            "filled": true,
+            "author_pub_id": "eRbvWqYAAAAJ:u5HHmVD_uO8C",
+            "num_citations": 99,
+            "pub_url": "http://books.google.com/books?hl=en&lr=&id=YCARBwAAQBAJ&oi=fnd&pg=PR7&dq=info:-qxGY874hFMJ:scholar.google.com&ots=-hU_2bXHcs&sig=GdbMwkW5-_-Q4ia6N0jZkNx7VPs",
+            "cites_id": "6018208567386352890",
+            "citedby_url": "/scholar?cites=6018208567386352890",
+            "url_related_articles": "/scholar?oi=bibs&hl=en&q=related:-qxGY874hFMJ:scholar.google.com/",
+            "cites_per_year": {
+                "2005": 4,
+                "2006": 5,
+                "2007": 3,
+                "2008": 6,
+                "2009": 5,
+                "2010": 7,
+                "2011": 6,
+                "2012": 5,
+                "2013": 6,
+                "2014": 6,
+                "2015": 4,
+                "2016": 5,
+                "2017": 9,
+                "2018": 5,
+                "2019": 10,
+                "2020": 7
+            },
+            "eprint_url": "https://www.jugglingedge.com/pdf/juggling_survey_Burkard_Polster.pdf"
+        }
+    ]}`)["publications"];*/
+    console.log(publicationList);
+    let returnList = []
+    for(let i =0;i<publicationList.length;i++){
+        let currentPub = publicationList[i];
+        let pubexport = new Object();
+        pubexport["authors"] = [currentPub["bib"]["author"]];
+        pubexport["title"] = currentPub["bib"]["title"];
+        pubexport["description"] = currentPub["bib"]["abstract"];
+        pubexport["year_published"] = currentPub["bib"]["pub_year"];
+        pubexport["link"] = currentPub["eprint_url"];
+        pubexport["citedBy"] = currentPub["num_citations"];
+        returnList.push(pubexport);
+        //let pub = await Publication.create(i);
+        //console.log(pub);
+    }
+    res.status(200).json(returnList);
+
+}
+module.exports = {deletePublication, updatePublication, createPublication, readPublication, readAllPublicationsByTeam, importPublication};
```

Comment:
Can you add a try catch statement to catch the error if the authorId is invalid?

Reply:
I believe this just gives you an empty json, not an error